test/e2e/specs/editor/various/multi-block-selection.spec.js

```diff
@@ -0,0 +1,1372 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+/** @typedef {import('@playwright/test').Page} Page */
+/** @typedef {import('@wordpress/e2e-test-utils-playwright').Editor} Editor */
+
+/**
+ * Some tests in this file use the character `|` to represent the caret's position
+ * in a more readable format.
+ */
+test.describe( 'Multi-block selection', () => {
+	test.use( {
+		multiBlockSelectionUtils: async ( { page, editor }, use ) => {
+			await use( new MultiBlockSelectionUtils( { page, editor } ) );
+		},
+	} );
+
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'should select with double ctrl+a and speak', async ( {
+		page,
+		editor,
+		pageUtils,
+		multiBlockSelectionUtils,
+	} ) => {
+		for ( let i = 1; i <= 3; i += 1 ) {
+			await editor.insertBlock( {
+				name: 'core/paragraph',
+				attributes: { content: `${ i }` },
+			} );
+		}
+
+		// Multiselect via keyboard.
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+
+		await expect
+			.poll( multiBlockSelectionUtils.getSelectedFlatIndices )
+			.toEqual( [ 1, 2, 3 ] );
+
+		await expect( page.locator( '[aria-live="assertive"]' ) ).toHaveText(
+			'3 blocks selected.'
+		);
+	} );
+
+	// See #14448: an incorrect buffer may trigger multi-selection too soon.
+	test( 'should only trigger multi-selection when at the end', async ( {
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		// Create a paragraph with four lines.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: '1.<br>2.<br>3.<br>4.' },
+		} );
+		// Create a second block and focus it.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: '' },
+		} );
+		await editor.canvas
+			.getByRole( 'document', { name: /Empty block/i } )
+			.click();
+		// Move to the middle of the first line.
+		await pageUtils.pressKeyTimes( 'ArrowUp', 4 );
+		await page.keyboard.press( 'ArrowRight' );
+		// Select mid line one to mid line four.
+		await pageUtils.pressKeyTimes( 'Shift+ArrowDown', 3 );
+		// Delete the text to see if the selection was correct.
+		await page.keyboard.press( 'Backspace' );
+
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{ name: 'core/paragraph', attributes: { content: '1.' } },
+			{ name: 'core/paragraph', attributes: { content: '' } },
+		] );
+	} );
+
+	test( 'should use selection direction to determine vertical edge', async ( {
+		page,
+		editor,
+	} ) => {
+		await editor.canvas
+			.getByRole( 'button', { name: 'Add default block' } )
+			.click();
```

Comment:
Any reason we're not inserting the block programmatically here as well?

```suggestion
		await editor.insertBlock( {
			name: 'core/paragraph',
			attributes: { content: '' },
```

Reply:
Because `insertBlock` doesn't place the caret at the end of the paragraph block. It's just a bit easier sometimes to manually type it than focusing and moving the caret after inserting the block.